### PR TITLE
[WIP-docs] Accelerate scripts

### PR DESCRIPTION
**Pull request by stevhliu**
This PR explores what a separate section for the Accelerate integrations (DeepSpeed + FSDP) could look like. The main parts I wanted to focus on are:

- generalize the DeepSpeed script a bit more so users can configure it with other settings besides ZeRO-3/Offload
- parts of the script users can customize if they want to use a different model, dataset, or PEFT method
- highlight and briefly explain what the Accelerate parts are doing

~If we think this makes sense, I'll also do the FSDP script~ done :)

### CODE REVIEW

File: docs/source/accelerate/deepspeed-zero3-offload.mdx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+```
+
+That is all! The rest of the script handles the training loop, evaluation, and even pushes it to the Hub for you.
+
+## Train
+
+Run the following command to launch the training script:
+
+```bash
+accelerate launch --config_file ds_zero3_cpu.yaml examples/peft_lora_seq2seq_accelerate_ds_zero3_offload.py
```
Comment (MKhalusova) [inline docs/source/accelerate/deepspeed-zero3-offload.mdx:128]: Here the config has a name. How do we name a config when creating it with `accelerate config`? 
Maybe if we show it at the beginning, it will be more clear that here we are using the config that was created earlier in the guide.

**Review by MKhalusova — COMMENTED** [review]
I like the approach you take in this example to make it more generalized and customizable! Given so many moving parts (peft, accelerate, deepspeed, script, and config), I've added a few suggestions to improve clarity.

File: docs/source/accelerate/deepspeed-zero3-offload.mdx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+```bash
+`zero_stage`: [0] Disabled, [1] optimizer state partitioning, [2] optimizer+gradient state partitioning and [3] optimizer+gradient+parameter partitioning
+`gradient_accumulation_steps`: Number of training steps to accumulate gradients before averaging and applying them.
+`gradient_clipping`: Enable gradient clipping with value.
+`offload_optimizer_device`: [none] Disable optimizer offloading, [cpu] offload optimizer to CPU, [nvme] offload optimizer to NVMe SSD. Only applicable with ZeRO >= Stage-2.
+`offload_param_device`: [none] Disable parameter offloading, [cpu] offload parameters to CPU, [nvme] offload parameters to NVMe SSD. Only applicable with ZeRO Stage-3.
+`zero3_init_flag`: Decides whether to enable `deepspeed.zero.Init` for constructing massive models. Only applicable with ZeRO Stage-3.
+`zero3_save_16bit_model`: Decides whether to save 16-bit model weights when using ZeRO Stage-3.
+`mixed_precision`: `no` for FP32 training, `fp16` for FP16 mixed-precision training and `bf16` for BF16 mixed-precision training.
```
Comment (pacman100) [inline docs/source/accelerate/deepspeed-zero3-offload.mdx:33]: nice explanation!

**Review by pacman100 — APPROVED** [review]
Thank you @stevhliu for adding the accelerate script examples. Great overviews of DeepSpeed/FSDP, their config parameters, pointing out the major changes in the scripts related to PEFT/DeepSpeed/FSDP. Thank you for all the work, LGTM! 🤗🚀✨
